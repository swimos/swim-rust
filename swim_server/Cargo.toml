[package]
name = "swim_server"
version = "0.1.0"
authors = ["Swim.AI developers info@swim.ai"]
edition = "2018"

[features]
default = []
debug = ["log_verbose"]
log_verbose = []
trust-dns = ["trust-dns-resolver"]

[target.'cfg(log_verbose)'.dependencies]
tracing = { version = "0.1.13", features = ["max_level_debug"] }

[target.'cfg(not(log_verbose))'.dependencies]
tracing = "0.1.16"

[dependencies]
either = "1.5.3"
futures = "0.3.4"
futures-util = "0.3.4"
http = "0.2.3"
im = "15.0.0"
parking_lot = "0.11.0"
pin-project = "1.0.5"
pin-utils = "0.1.0"
tokio = { version = "1.1.1", features = ["rt", "net"] }
tokio-stream = "0.1.2"
url = "2.1.1"
swim_common = { path = "../swim_common", features = ["tls"] }
swim_warp = { path = "../swim_warp" }
utilities = { path = "../utilities" }
swim_runtime = { path = "../swim_runtime" }
stm = { path = "../stm" }
tracing-futures = "0.2.4"
agent_derive = { path = "agent_derive" }
trybuild = "1.0.17"
tokio-tungstenite = { version = "0.13.0", git = "https://github.com/swimos/tokio-tungstenite/", branch = "deflate-v13", features = ["tls", "deflate"] }
tokio-native-tls = "0.3"
trust-dns-resolver = { version = "0.20", optional = true, features = ["tokio-runtime"] }
auth = { path = "auth" }
regex = "1"
lazy_static = "1.4.0"
percent-encoding = "2.1.0"
thiserror = "1.0"

[dev-dependencies]
tokio = { version = "1.1.1", features = ["rt-multi-thread", "macros", "test-util"] }
tracing = "0.1.16"
tokio-test = "0.4.0"
