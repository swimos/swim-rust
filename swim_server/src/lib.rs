// Copyright 2015-2021 Swim Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#![allow(clippy::match_wild_err_arm)]

pub mod agent;
pub mod interface;
pub mod meta;
#[macro_use]
pub mod macros;
pub mod routing;
mod scheduler;
pub mod sync {
    pub mod circular_buffer {
        pub use swim_utilities::sync::circular_buffer::{
            channel,
            error::{RecvError, SendError},
            watch_channel, Receiver, Sender,
        };
    }
}

mod plane;

#[allow(unused_imports)]
pub use agent_derive::*;
pub use plane::spec::PlaneBuilder;
pub use stm;
pub use stringify_attr::{stringify_attr, stringify_attr_raw};
pub use swim_utilities::future::retryable::RetryStrategy;
pub use swim_utilities::future::SwimStreamExt;
pub use swim_utilities::routing::route_pattern::RoutePattern;
pub use swim_utilities::routing::route_uri;

#[doc(hidden)]
pub mod store {
    pub use swim_persistence::agent::lane::value::ValueLaneStoreIo;
    pub use swim_persistence::agent::lane::{LaneNoStore, StoreIo};
    pub use swim_persistence::agent::NodeStore;
}
