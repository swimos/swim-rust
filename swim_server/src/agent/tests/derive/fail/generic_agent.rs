use swim_server::SwimAgent;

fn main() {
    #[derive(Debug)]
    pub struct TestAgentConfig;

    #[derive(Debug, SwimAgent)]
    #[agent(config = "TestAgentConfig")]
    pub struct TestAgent<S> {
        field: S,
    }
}
