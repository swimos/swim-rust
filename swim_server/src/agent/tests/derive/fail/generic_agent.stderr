error: Swim agent cannot have generic parameters.
  --> $DIR/generic_agent.rs:22:5
   |
22 | /     #[agent(config(TestAgentConfig))]
23 | |     pub struct TestAgent<S> {
24 | |         field: S,
25 | |     }
   | |_____^
